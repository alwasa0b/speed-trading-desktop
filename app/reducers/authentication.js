import { LOGIN_SUCCESS } from "../constants/login";

export default (state = { loggedIn: false }, action) => {
  switch (action.type) {
    case LOGIN_SUCCESS:
      return {
        loggedIn: true
      };
    default:
      return state;
  }
};
