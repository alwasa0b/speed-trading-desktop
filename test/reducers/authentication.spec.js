import authentication from "../../app/reducers/authentication";

describe("reducers", () => {
  describe("authentication", () => {
    it("should handle initial state", () => {
      expect(authentication(undefined, {})).toMatchSnapshot();
    });

    it("should handle LOGIN_SUCCESS", () => {
        expect(authentication(undefined, {type:"LOGIN_SUCCESS"})).toMatchSnapshot();
      });
  });
});
