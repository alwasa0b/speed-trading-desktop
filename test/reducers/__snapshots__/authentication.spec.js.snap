// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers authentication should handle LOGIN_SUCCESS 1`] = `
Object {
  "loggedIn": true,
}
`;

exports[`reducers authentication should handle initial state 1`] = `
Object {
  "loggedIn": false,
}
`;
